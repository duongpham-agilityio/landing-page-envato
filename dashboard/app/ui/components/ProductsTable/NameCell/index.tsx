// Libs
import { memo } from 'react';
import { Flex, Td, Text, Tooltip } from '@chakra-ui/react';

interface NameCellProps {
  name: string;
}

const NameCell = ({ name }: NameCellProps): JSX.Element => (
  <Td
    py={5}
    pr={5}
    pl={0}
    fontSize="md"
    color="text.primary"
    fontWeight="semibold"
    textAlign="left"
    w={{ base: 200, xl: 220, '3xl': 200, '6xl': 250 }}
  >
    <Flex alignItems="center" gap="10px">
      <Tooltip minW="max-content" placement="bottom-start" label={name}>
        <Text
          display="block"
          fontSize="md"
          fontWeight="semibold"
          wordBreak="break-all"
          textOverflow="ellipsis"
          overflow="hidden"
          pr={10}
          flex={1}
          w={{ base: 200, xl: 220, '3xl': 200, '6xl': 250 }}
        >
          {name}
        </Text>
      </Tooltip>
    </Flex>
  </Td>
);

const NameCellMemorized = memo(NameCell);

export default NameCellMemorized;
