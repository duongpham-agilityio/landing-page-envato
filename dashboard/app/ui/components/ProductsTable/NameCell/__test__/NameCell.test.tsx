import { render } from '@testing-library/react';
import { Table } from '@chakra-ui/react';

// Components
import NameCell from '..';

const mockProps = {
  name: 'shoe',
};

describe('NameCell component', () => {
  const setup = () =>
    render(<NameCell {...mockProps} />, {
      wrapper: Table,
    });

  it('should match with snapshot', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
});
