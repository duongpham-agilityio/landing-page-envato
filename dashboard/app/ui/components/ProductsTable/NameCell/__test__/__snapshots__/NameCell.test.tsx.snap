// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameCell component should match with snapshot 1`] = `
<div>
  <table
    class="chakra-table css-0"
  >
    <td
      class="css-xpdvke"
    >
      <div
        class="css-1h2sv0j"
      >
        <p
          class="chakra-text css-s51d82"
        >
          shoe
        </p>
      </div>
    </td>
  </table>
</div>
`;
